pkg/consoleplugin/consolepluginserver.go

```diff
@@ -0,0 +1,69 @@
+package consoleplugin
+
+import (
+	"context"
+	"k8s.io/utils/strings/slices"
+	"net/http"
+	"os"
+
+	"github.com/codeready-toolchain/member-operator/pkg/consoleplugin/scriptserver"
+
+	"github.com/go-logr/logr"
+)
+
+const (
+	ConsolePluginServerOptionNoTLS = ConsolePluginServerOption("notls")
+)
+
+type ConsolePluginServerOption string
+
+type Server struct {
+	mux     *http.ServeMux
+	svr     *http.Server
+	log     logr.Logger
+	options []string
+}
+
+func NewConsolePluginServer(log logr.Logger, options ...ConsolePluginServerOption) *Server {
+	s := &Server{
+		log: log,
+	}
+
+	for _, opt := range options {
+		s.options = append(s.options, string(opt))
+	}
+
+	s.mux = http.NewServeMux()
+	ss := scriptserver.NewScriptServer()
+	s.mux.HandleFunc("/", ss.HandleScriptRequest)
+	s.svr = &http.Server{ //nolint:gosec
+		Addr:    ":9443",
```

Comment:
small detail - should this port to be configured ? maybe setting it from env variables could make it more flexible.

Reply:
It's hard to make it truly configurable because the port is also defined in the Service manifest. I think it's fine to hardcode it.